tensordict/functional.py

```diff
@@ -77,7 +77,7 @@ def pad(tensordict: T, pad_size: Sequence[int], value: float = 0.0) -> T:
 
 def pad_sequence(
     list_of_tensordicts: Sequence[T],
-    batch_first: bool = True,
+    pad_dim: int = 0,
```

Comment:
This is bc breaking, we should keep batch_first as the last learn and set its default to None. If it's not None we raise a deprecation warning.